corehq/apps/app_manager/views/formdesigner.py

```diff
@@ -53,216 +52,77 @@
-            {
-                'key': 'fixture',
-                'name': 'Fixtures',
-                'endpoint': reverse('fixture_metadata', kwargs={'domain': domain}),
-            },
-        ],
-        'form': form.source,
-        'formId': form.get_unique_id(),
-        'formName': trans(form.name, app.langs),
-        'saveType': 'patch',
-        'saveUrl': reverse('edit_form_attr', args=[domain, app.id, form.get_unique_id(), 'xform']),
-        'patchUrl': reverse('patch_xform', args=[domain, app.id, form.get_unique_id()]),
-        'allowedDataNodeReferences': [
-            "meta/deviceID",
-            "meta/instanceID",
-            "meta/username",
-            "meta/userID",
-            "meta/timeStart",
-            "meta/timeEnd",
-            "meta/location",
-        ] + scheduler_data_nodes,
-        'activityUrl': reverse('ping'),
-        'sessionid': request.COOKIES.get('sessionid'),
-        'externalLinks': {
-            'changeSubscription': reverse("domain_subscription_view", kwargs={'domain': domain}),
-        },
-        'invalidCaseProperties': ['name'],
-    }
-
-    if form.get_action_type() == 'open':
-        core.update({
-            'defaultHelpTextTemplateId': '#fd-hq-helptext-registration',
-            'formIconClass': 'fcc fcc-app-createform',
-        })
-    elif form.get_action_type() == 'close':
-        core.update({
-            'defaultHelpTextTemplateId': '#fd-hq-helptext-close',
-            'formIconClass': 'fcc fcc-app-completeform',
-        })
-    elif form.get_action_type() == 'update':
-        core.update({
-            'defaultHelpTextTemplateId': '#fd-hq-helptext-followup',
-            'formIconClass': 'fcc fcc-app-updateform',
-        })
-    else:
-        core.update({
-            'defaultHelpTextTemplateId': '#fd-hq-helptext-survey',
-            'formIconClass': 'fa fa-file-o',
-        })
-
-    vellum_options = {
-        'core': core,
-        'plugins': vellum_plugins,
-        'features': vellum_features,
-        'intents': {
-            'templates': next(app_callout_templates),
-        },
-        'javaRosa': {
-            'langs': app.langs,
-            'displayLanguage': context['lang'],
-        },
-        'uploader': {
-            'uploadUrls': {
-                'image': reverse("hqmedia_uploader_image", args=[domain, app.id]),
-                'audio': reverse("hqmedia_uploader_audio", args=[domain, app.id]),
-                'video': reverse("hqmedia_uploader_video", args=[domain, app.id]),
-                'text': reverse("hqmedia_uploader_text", args=[domain, app.id]),
-            },
-            'objectMap': app.get_object_map(),
-            'sessionid': request.COOKIES.get('sessionid'),
-        },
-    }
-    context.update({
-        'vellum_options': vellum_options,
-        'CKEDITOR_BASEPATH': "app_manager/js/vellum/lib/ckeditor/",
+        'can_preview_form': request.couch_user.has_permission(
```

Comment:
I don't think this is used in templates anymore.

Reply:
perf thanks